Marlin/src/gcode/geometry/G50_G51.cpp

```diff
@@ -0,0 +1,86 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2020 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "../../inc/MarlinConfigPre.h"
+
+#if ENABLED(SCALE_WORKSPACE)
+
+  #include "../gcode.h"
+  #include "../../module/motion.h"
+
+  /**
+   * G50: Set Workspace Scaling
+   *
+   * Scale the current workspace coordinate system.
+   *
+   * Parameters:
+   *   X<linear>  X coordinate of the scaling center
+   *   Y<linear>  Y coordinate of the scaling center
+   *   Z<linear>  Z coordinate of the scaling center
+   *   I<float>   scaling factor for X axis
+   *   J<float>   scaling factor for Y axis
+   *   K<float>   scaling factor for Z axis
+   *   P<float>   scaling factor
+   */
+  void GcodeSuite::G50() {
```

Comment:
G51 should enable scaling, G50 should cancel scaling, as demonstrated in my PR. Can you switch it back, please? Source:

Haas: https://www.haascnc.com/service/online-operator-s-manuals/mill-operator-s-manual/mill---g-codes.html#gsc.tab=0

Fanuc: https://www.fanucamerica.com/docs/default-source/cnc-files/brochures/function-catalog.pdf?sfvrsn=999f41bd_13 , 
https://www.cnccookbook.com/g51-scaling-the-cnc-coordinate-system/